dask/bytes/s3.py

```diff
@@ -200,3 +211,51 @@ def getsize(path, compression, s3):
 core._read_bytes['s3'] = read_bytes
 core._open_files['s3'] = open_files
 core._open_files_write['s3'] = open_file_write_direct
+
+
+class DaskS3FileSystem(S3FileSystem):
+
+    def __init__(self, key=None, username=None, secret=None, password=None,
+                 blocksize=5 * 2 ** 20, **kwargs):
+        if username is not None:
+            if key is not None:
+                raise KeyError("S3 storage options got secrets argument "
+                               "collision. Please, use either `key` "
+                               "storage option or password field in URLpath, "
+                               "not both options together.")
+        key = username
+        if key is not None:
+            kwargs['key'] = key
+        if password is not None:
+            if secret is not None:
+                raise KeyError("S3 storage options got secrets argument "
+                               "collision. Please, use either `secret` "
+                               "storage option or password field in URLpath, "
+                               "not both options together.")
+            secret = password
+        if secret is not None:
+            kwargs['secret'] = secret
+        self.blocksize = blocksize
+        S3FileSystem.__init__(self, kwargs)
+
+    def open(self, path, mode='rb', **kwargs):
+        bucket = kwargs.pop('host', '')
+        s3_path = bucket + path
+        return S3FileSystem.open(self, s3_path, mode=mode,
+                                 block_size=self.blocksize)
+
+    def glob(self, path, **kwargs):
+        bucket = kwargs.pop('host', '')
+        s3_path = bucket + path
+        return S3FileSystem.glob(self, s3_path)
+
+    def makedirs(self, path):
+        pass  # no need to pre-make paths on S3
+
+    def ukey(self, path):
+        return self.info(path)['ETag']
+
+    def size(self, path):
+        return self.info(path)['Size']
+
+core._filesystems['s3'] = DaskS3FileSystem
```

Comment:
It would be nice if the FileSystem objects in s3fs and hdfs3 could satisfy the necessary contract on their own, without wrapper classes.  Is this doable?